### PR TITLE
Revised behavior on ill-formed identifiers

### PR DESCRIPTION
**Pull request by herbelin**
This is the first half of #618 (proposal for clarifying the status of the invalid-identifier).

### CODE REVIEW

File: kernel/names.ml [hunk 1]
```diff
@@ -34,16 +34,8 @@ struct
...
-    CWarnings.create ~name:"invalid-identifier" ~category:"parsing"
-                     ~default:CWarnings.Disabled
-                     (fun s -> str s)
-
-  let check_soft ?(warn = true) x =
-    let iter (fatal, x) =
-      if fatal then CErrors.error x else
-        if warn then warn_invalid_identifier x
-    in
+  let check_soft ?(strict=true) x =
```
Comment (maximedenes) [inline kernel/names.ml]: Shouldn't this be called `check_identifier`? I thought `soft` meant that we try not to fail when possible, which is not the case here if `strict=true`.

Comment (herbelin) [inline kernel/names.ml]: It used to be called `check_ident_soft` but it was renamed to `Id.check_soft` in 67f5c70a (PMP, Dec 2012). So, to be consistent with the qualified name view, maybe `check_valid`?

BTW, I realize that the comments in the mli have to be updated. I'm going to do it.

Comment (maximedenes) [inline kernel/names.ml]: No objection to `check_valid` my point is that `soft` should not stay in the name if that is controllable through a flag.

**Comment by maximedenes** [comment]
LGTM, thanks!

**Pull request merged**